README.Docker.md

```diff
@@ -0,0 +1,246 @@
+docker. This section briefly describes them, but the other sections of this
+document are best read before hand to better understand some of the concepts.
+
+One important aspect is however to realize, that docker by default does not have
+access to any of the hosts files. And so to scan these within docker, they need
+to be volume mounted to be made accessible.
+```console
+       --volume '/path/to/scan:/scandir'
+```
+would make the hosts file/directory `/path/to/scan` available in the container
+as `/scandir` and thus invoking clamscan would thus be done on `/scandir`.
+
+Note that while technically possible to run either scanners via `docker exec`
+this is not described as it is unlikely the container has access to the files
+to be scanned.
+
+
+### clamscan
+Using `clamscan` outside of the docker container is how normally clamscan is
+invoked. To make use of the available shared dockerized resources however, it
+is possible to expose the virus database and share that for example. E.g. it
+could be possible to run a docker container with only `freshclamd` and share
+the virus database directory `/usr/share/clamav`. This could be useful for file
+servers for example, where only `clamscan` is installed on the host, and
+`freshlamd` is managed in a docker container.
+
+Note, that running `freshclamd` separated from `clamd` is less recommended,
+unless the `clamd` socket is shared with `freshclamd` as `freshclamd` would not
+be able to inform `clamd` of database updates.
+
+
+### Dockerized clamscan
+To run `clamscan` in a docker container, the docker container can be invoked
+as:
+```console
+docker run \
+       --interactive \
+       --rm \
+       --tty \
+       --volume '/path/to/scan:/scandir' \
+       "myclamav" clamscan /scandir
+```
+This would pull the virus database using `freshclam`, run the scan and cleanup.
+
+Note, that this is very inefficient when done often, as the complete database is
+downloaded on each invocation.
+
+
+### clamdscan
+As with `clamscan` `clamdscan` can also be run when installed on the host, by
+connecting to the dockerized `clamd`. This can be done by either pointing
+`clamdscan` to the exposed TCP/UDP port or unix socket.
+
+
+### Dockerized clamdscan
+Running both `clamd` and `clamdscan` is also easily possible, as all that is
+needed is the shared socket between the two containers. The only cavaet here
+is to mount the files to be scaned in the container that is expected to run
+`clamdscan`.
+
+For example:
+```console
+docker run \
+       --interactive \
+       --rm \
+       --tty \
+       --volume '/path/to/scan:/scandir' \
+       --volume '/var/lib/docker/data/clamav/sockets/:/run/clamav/'
+       "myclamav" clamdscan /scandir
+```
+
+
+## Controlling the container
+The ClamAV container actually runs both `freshclam` and `clamd` by default.
+Optionally available to the container is also ClamAV's milterd. To control the
```

Comment:
Please change this and other references from `milterd` to `clamav-milter` or just "milter" (for casual references instead of actual commands).  

`clamav-milterd` and `clamonaccd` might've been better names, but sadly we don't have the `d` suffix.

Reply:
will do